### ISSUE TITLE
Support truffle boxes where the truffle project is not at the project root (e.g. monorepos)

### ISSUE THREAD

**Issue by adrianmcli** [trufflesuite/truffle-box#11]
The current implementation forces all valid truffle boxes to have the truffle project be in the base directory for the repo by checking for a `/master/truffle.js`.

```js
    var configURL = parseURL(
      vcsurl(url)
        .replace("github.com", "raw.githubusercontent.com")
        .replace(/#.*/, "") +
        "/master/truffle.js"
);
```

[Source](https://github.com/trufflesuite/truffle-box/blob/ceaa439503ca43d8eea23b93f149dc2af90ec091/lib/utils/unbox.js#L33)

I am the author of a Truffle Box ([link](https://github.com/adrianmcli/truffle-next)) and I wanted to convert my Truffle Box into a monorepo instead. By that I mean I wanted to separate the frontend client and truffle project into two separate folders (not nested): `/truffle` and `/client`.

In other words, I wanted to be able to run `truffle unbox <my truffle box name>` and immediately get a monorepo of a truffle project and my frontend client in separate folders. However, the current implementation does not allow that and I understand why that might be the case. I assume you want all truffle boxes to deposit the truffle project in the current directory.

Nevertheless, I thought this was worth an issue anyhow.

**Comment by gasolin** [comment]
+1, `/master/truffle.js` should not be essential at the root of  the template, but `truffle-box.json` does.

https://github.com/trufflesuite/truffle-box/blob/develop/lib/utils/unbox.js#L36
